packages/core/src/transport/httpRetry.ts

```diff
@@ -0,0 +1,133 @@
+import type { TimeoutId } from '../tools/utils'
+import { ONE_SECOND } from '../tools/utils'
+import type { Observable } from '../tools/observable'
+import { monitor } from '../tools/monitor'
+
+const MIN_RETRY_INTERVAL = ONE_SECOND
+const MAX_RETRY_INTERVAL = 8 * ONE_SECOND
+
+export function httpRetry(
+  send: (data: string | FormData, bytesCount: number) => Observable<Response>,
```

Comment:
In this PR I assume `send` already uses fetch keepalive and return an `Observable`